### PR TITLE
storage: Improve raid degraded message

### LINKED ISSUE #3960: storage: RAID failure message uses caps
If I remove one disk from my RAID I get the message:
"ARRAY IS DEGRADED -- 1 disks are missing"

Even though this is a pretty severe error that can result in data loss, I think it's best to avoid uppercap letters. It reads badly because the grammatical number is wrong (disks vs. disk)
1 disk, 2 disks, 3 disks.

How about something like:
"1 disk has failed or is missing, so the RAID Array is in a degregated state"

### PR DESCRIPTION
**Pull request by dperpeet**
Fixes #3960 
Follow-up to #3975

### CODE REVIEW

**Comment by dperpeet** [comment]
screenshot from 2016-03-30 15-51-50

**Comment by dperpeet** [comment]
my typo commit can be squashed into andreas' commit

**Comment by dperpeet** [comment]
I need to change the tests.

**Comment by dperpeet** [comment]
@andreasn can you take a look at the css please?

**Comment by andreasn** [comment]
I wonder if it makes more sense to put it above the panel, since it affects the whole page, including the disks.
(one feedback we got in the past was that someone thought only the things within the RAID Device panel belonged to the RAID)

Somewhat like this:
screenshot from 2016-03-31 11-57-08

This would allow us to save up on some of the special case css for tables.

**Comment by dperpeet** [comment]
screenshot from 2016-03-31 21-22-08
You're right, this looks better. Implemented the change, also fewer extra CSS now.

@andreasn As a follow-up, you should look at the alert padding in cockpit.css. It seems like the bottom should be more like 7-8px instead of 5 like the top.

**Comment by dperpeet** [comment]
I also changed the warning text depending on the number of disks: "1 disk is missing" or "n disks are missing" (n>1).

**Comment by dperpeet** [comment]
tests still need fixing

**Comment by andreasn** [comment]
> I also changed the warning text depending on the number of disks: "1 disk is missing" or "n disks are missing" (n>1).

Thanks! I was thinking about that the other day too. Until it can give different messages for singular and plural, opting for (what I hope is) the more common situation of just one broken disk is better.

**Comment by andreasn** [comment]
> As a follow-up, you should look at the alert padding in cockpit.css. It seems like the bottom should be more like 7-8px instead of 5 like the top.

Sure, I'll take a look and file a separate issue.

**Comment by dperpeet** [comment]
> Thanks! I was thinking about that the other day too. Until it can give different messages for singular and plural, opting for (what I hope is) the more common situation of just one broken disk is better.

The message does depend on the number of disks.

**Comment by andreasn** [comment]
> The message does depend on the number of disks.

You're right. I read that, but at the same time, I didn't read it apparently :)

File: pkg/storaged/details.js [hunk 1]
```diff
@@ -1419,13 +1419,21 @@ define([
                 };
             }
 
+            var degraded_message = null;
+            if (mdraid.Degraded > 0) {
+                if (mdraid.Degraded > 1)
```
Comment (mvollmer) [inline pkg/storaged/details.js]: This should use `cockpit.ngettext`.  See https://www.gnu.org/software/gettext/manual/html_node/Plural-forms.html for a lecture.

Comment (dperpeet) [inline pkg/storaged/details.js]: > This should use cockpit.ngettext.

Done. I didn't even know we had that function...

**Pull request closed without merging**